be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs

```diff
@@ -864,20 +864,23 @@ private async Task MapStatusesToOrderStatusId(Guid? orderId)
                         lastOrderItemStatusChanges.Add(new OrderItemStatusServiceModel
                         {
                             OrderItemStateId = lastOrderItemStatusChange.OrderItemStateId,
-                            OrderItemStatusId = lastOrderItemStatusChange.OrderItemStatusId
+                            OrderItemStatusId = lastOrderItemStatusChange.OrderItemStatusId,
+                            CreatedDate = lastOrderItemStatusChange.CreatedDate
                         });
                     }
                 }
 
                 bool isSameStatus = lastOrderItemStatusChanges.DistinctBy(x => x.OrderItemStatusId).Count() == 1;
 
+                var lastOrderItemStatus = lastOrderItemStatusChanges.FirstOrDefault();
```

Comment:
**suggestion:** let's add a null check for lastOrderItemStatus . If it's null there will be an exception on line 879-880.